post-docs/add-back-references.py

```diff
@@ -76,30 +88,49 @@ def create_back_reference_html(back_ref_url, path):
         f.write(content)
 
 
-download_file(docs_link)
-old_to_new = construct_mapping()
-
-versions = [f.path.split("/")[-1] for f in os.scandir(apache_airflow_path) if f.is_dir()]
-versions = [v for v in versions if version_is_less_than(v)]
-
-for version in versions:
-    r = apache_airflow_path + "/" + version
-
-    for p in old_to_new:
-        old = p
-        new = old_to_new[p]
-
-        # only if old file exists, add the back reference
-        if os.path.exists(r + "/" + p):
-            d = old_to_new[p].split("/")
-            file_name = old_to_new[p].split("/")[-1]
-            dest_dir = r + "/" + "/".join(d[: len(d) - 1])
-
-            # finds relative path of old file wrt new, handles case of different file names too
-            relative_path = os.path.relpath(old, new)
-            # remove one directory level because file path was used above
-            relative_path = relative_path.replace("../", "", 1)
-
-            os.makedirs(dest_dir, exist_ok=True)
-            dest_file_path = dest_dir + "/" + file_name
-            create_back_reference_html(relative_path, dest_file_path)
+def generate_back_references(link, base_path, change_version):
+    download_file(link)
+    old_to_new = construct_mapping()
+
+    versions = [f.path.split("/")[-1] for f in os.scandir(base_path) if f.is_dir()]
+    versions = [v for v in versions if version_is_less_than(v, change_version)]
+
+    for version in versions:
+        r = base_path + "/" + version
+
+        for p in old_to_new:
+            old = p
+            new = old_to_new[p]
+
+            # only if old file exists, add the back reference
+            if os.path.exists(r + "/" + p):
+                d = old_to_new[p].split("/")
+                file_name = old_to_new[p].split("/")[-1]
+                dest_dir = r + "/" + "/".join(d[: len(d) - 1])
+
+                # finds relative path of old file wrt new, handles case of different file names too
+                relative_path = os.path.relpath(old, new)
+                # remove one directory level because file path was used above
+                relative_path = relative_path.replace("../", "", 1)
+
+                os.makedirs(dest_dir, exist_ok=True)
+                dest_file_path = dest_dir + "/" + file_name
+                create_back_reference_html(relative_path, dest_file_path)
+
+
+# total arguments
+n = len(sys.argv)
+if n != 2:
+    logging.Logger.error("missing required arguments, syntax: python add-back-references.py [airflow | providers | "
+                         "helm]")
+
+type_generation = sys.argv[1]
+if type_generation == airflow_type:
+    generate_back_references(airflow_redirects_link, airflow_docs_path, new_airflow_docs_version)
+elif type_generation == helm_type:
+    generate_back_references(helm_redirects_link, helm_docs_path, new_helm_docs_version)
+elif type_generation == provider_type:
+    pass
```

Comment:
Is it work in progress?

Reply:
Maybe for providers this does not apply, a log explaining that could help ?